core/sql/optimizer/ValueDesc.cpp

```diff
@@ -3165,7 +3165,11 @@ void ValueIdSet::replaceVEGExpressions
           if (iePtr != exprId.getItemExpr())  // a replacement was done
 	    {
 	      subtractElement(exprId);        // remove existing ValueId
-	      newExpr += iePtr->getValueId(); // replace with a new one
+	      //insert new expression(s)
+	      if (iePtr->getOperatorType() == ITM_AND)
+	    	  iePtr->convertToValueIdSet(newExpr, NULL, ITM_AND, FALSE, FALSE);
```

Comment:
Should add a comment to ValueIdSet::replaceVEGExpressions() on converting the AND backbone into a valueId set.